Add ccode for FloorDiv

kalpit-meta-1 opened the pull request:
Summary: Add ccode for FloorDiv

Test Plan: CIs

Differential Revision: D70749021




cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

isuruf commented on torch/utils/_sympy/functions.py:
```diff
@@ -291,6 +291,11 @@ def eval(
+    def _ccode(self, printer):
+        x, y = self.args
+        _x, _y = printer._print(x), printer._print(y)
+        return f"floor({_x}/{_y})"
```
Don't you need parenthesis like `({_x})`?

Also is there a test for this? I don't think we use the ccode inside pytorch

kalpit-meta-1 replied:
Updated the code to use parenthesis similar to _sympystr

kalpit-meta-1 replied:
The use case for this is when export creates floordiv, we want to use sympy.ccode for it. Since this symbol has no direct mapping, hence adding it here. Pytorch will not use this ccode
I am not able to find a correct place for adding this specific unit test, do you have a place in mind?

kalpit-meta-1 committed:
Add ccode for FloorDiv (#148727)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/148727

Add ccode for FloorDiv

Test Plan: CIs

Differential Revision: D70749021

kalpit-meta-1 commented:
@pytorchbot merge
